### PR TITLE
Symbolic shapes mega merge PR (Oct 3)

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86160

- TensorGeometry supports symint
- check_size supports symint
- functorch batch rule improved symint
- Some operator support for symint in LTC
- More supported operations on SymInt and SymFloat
- More symint support in backwards formulas

This merge includes code contributions from bdhirsh and anjali411.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: torch/fx/experimental/symbolic_shapes.py
Comment (Chillee) [inline torch/fx/experimental/symbolic_shapes.py:502]: Don't land this, I think.

File: torch/fx/experimental/symbolic_shapes.py
Comment (Chillee) [inline torch/fx/experimental/symbolic_shapes.py:529]: I think this also slows down tracing quite a bit.

File: torch/fx/experimental/symbolic_shapes.py
Comment (ezyang) [reply torch/fx/experimental/symbolic_shapes.py:502]: I'm going to revert this from branch too then

File: torch/fx/experimental/symbolic_shapes.py
Comment (Chillee) [inline torch/fx/experimental/symbolic_shapes.py:181]: Why can't we use this here? https://docs.sympy.org/latest/modules/functions/elementary.html

Comment (ezyang) [reply torch/fx/experimental/symbolic_shapes.py:181]: ^- @bdhirsh

**Review by Chillee — APPROVED** [review]
symbolic shapes change is unclear to me, but will approve to unblock.

File: torch/fx/experimental/symbolic_shapes.py
Comment (bdhirsh) [reply torch/fx/experimental/symbolic_shapes.py:181]: Oh yeah we should use ‘ceiling’, I must have missed it. I mostly just copied off of how your did floor div

**Pull request closed without merging**
